### DISCUSSION TITLE
nnn, ranger, lf, joshuto, yazi, which is your choice?

### DISCUSSION THREAD

**Discussion by superiums** [kamiyaa/joshuto#454] [General]
# nnn, ranger, lf, joshuto, yazi, which is your choice?

only personal opionon:

## memery usage

in my case:
- nnn: 3.4MB
- joshuto: 5.4MB
- lf: 12.4MB
- ranger: 28MB
- yazi: 38MB

## funtion
- nnn
great turbo mode, fast to navigate. tons of plugins.

but the ui was simple, screen got a flash (redraw) if press `=`,

- lf
good documented
server mode instead of tabs.
seems no bookmark

- ranger
most carefully writed config files. ready to use. not matter you use which programs, it's almost considered in config.
most featured functions, very detailed. for example filter stack could fulfill your your logical.

- joshuto
most function and habit keeped from ranger.
more effetiontly. takes less memery.

- yazi
bulti preview makes it distinct from others.
precache mechinism, which makes your preview faster, but more resource usage.

## choice

- if you prefer simple and fast, nnn and joshuto is.
- if you prefer fast appearance and not care resouce usage, yazi maybe yours.
- if you prefer feature most, ranger and joshuto is. of course sth specture maybe yazi has.
- if you prefer blance of feature and tinny, joshuto and lf is.

only personal opininon, maybe not very exact true.
what's your opinion?

**Comment by sxyazi** [comment]
👋 Hi, Yazi author here. Just came across this discussion, and I have some different thoughts:

The memory comparison method seems a bit unfair to Yazi. Yazi embeds image encoding/decoding and code highlighting, which undoubtedly requires more memory than other file managers.

After Yazi processes tasks like image encoding, it doesn't immediately return the allocated memory to the system, since this is an inefficient operation that can result in a lot of memory fragmentation. Instead, Yazi retains some memory for potential future use rather than reapplying to the operating system, making its memory usage "appear" larger.

The system task manager includes the already freed memory, contributing to this perception. To see Yazi's actual memory usage, you can use tools like [heaptrack](https://github.com/KDE/heaptrack). In comparison, other file managers call external programs to perform these tasks, so they aren't considered part of the file manager's own memory. When these tasks are completed, the process is destroyed, and the memory is returned to the operating system. However, this involves a complete process creation cycle on each preview.

**Reply by 2060490154** [reply]
I hope Yazi can become the best terminal file manager

**Comment by superiums** [comment]
> However, this involves a complete process creation cycle on each preview.

yes, agree!

thanks to the open source community and the people like you who make us so plent of choice, great!

**Comment by hiql** [comment]
I used all of them, they are all great!!! My final choice is Yazi :)
No matter what your choice is, just use it, memory is not the problem, believe me~~

**Comment by lain-projects** [comment]
I use lf because its simple and easy to navigate.
